optuna/integration/pytorch_lightning.py

```diff
@@ -39,6 +44,18 @@ def __init__(self, trial: optuna.trial.Trial, monitor: str) -> None:
 
         self._trial = trial
         self.monitor = monitor
+        self.is_ddp_backend = False
+
+    def on_init_start(self, trainer: Trainer) -> None:
+        self.is_ddp_backend = trainer.accelerator_connector.distributed_backend is not None
+        if self.is_ddp_backend:
+            if version.parse(pl.__version__) < version.parse("1.4.0"):
+                raise ValueError("PyTorch Lightning>=1.4.0 is required in DDP.")
```

Comment:
Sorry, could you explain why we need to PyTorch lightning v 1.4.0 or higher version?

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/pull/7916
DDP training on multi GPUs supports from pytorch-lightning version 1.4.0.

Reply:
Ah, I got it. The ddp with callback has worked as we expected since v.1.4.0. Thanks!